praw/models/reddit/rules.py

```diff
@@ -0,0 +1,419 @@
+                            `Unix Time`_.
+    ``description``         The description of the rule, if provided, otherwise
+                            a blank string.
+    ``kind``                The kind of rule. Can be ``"link"``, ``comment"``,
+                            or ``"all"``.
+    ``priority``            Represents where the rule is ranked. For example,
+                            the first rule is at priority ``0``. Serves as an
+                            index number on the list of rules.
+    ``short_name``          The name of the rule.
+    ``violation_reason``    The reason that is displayed on the report menu for
+                            the rule.
+    ======================= ===================================================
+
+
+    .. _Unix Time: https://en.wikipedia.org/wiki/Unix_time
+
+    """
+
+    STR_FIELD = "short_name"
+
+    @cachedproperty
+    def mod(self) -> _RuleModeration:
+        """Contain methods used to moderate rules.
+
+        To delete ``"No spam"`` from the subreddit ``"NAME"`` try:
+
+        .. code-block:: python
+
+            reddit.subreddit('NAME').rules['No spam'].mod.delete()
+
+        To update ``"No spam"`` from the subreddit ``"NAME"`` try:
+
+        .. code-block:: python
+
+            reddit.subreddit("NAME").removal_reasons["No spam"].mod.update(
+                description="Don't do this!",
+                violation_reason="Spam post")
+        """
+        return RuleModeration(self)
+
+    def __init__(
+        self,
+        reddit: Reddit,
+        subreddit: Optional[Subreddit] = None,
+        short_name: Optional[str] = None,
+        _data: Optional[Dict[str, str]] = None,
+    ):
+        """Construct an instance of the Rule object."""
+        if (short_name, _data).count(None) != 1:
+            raise ValueError("Either short_name or _data needs to be given.")
+        self.short_name = short_name
+        # Note: The subreddit parameter can be None, because the objector
+        # does not know this info. In that case, it is the responsibility of
+        # the caller to set the `subreddit` property on the returned value.
+        self.subreddit = subreddit
+        super().__init__(reddit, _data=_data)
+
+    def __getattribute__(self, attribute: str) -> Any:
+        """Get the value of an attribute."""
+        value = super().__getattribute__(attribute)
+        if attribute == "subreddit" and value is None:
+            raise ValueError(
+                "The Rule is missing a subreddit. File a bug report at PRAW."
+            )
+        return value
+
+    def _fetch(self):
+        for rule in self.subreddit.rules:
+            if rule.short_name == self.short_name:
+                self.__dict__.update(rule.__dict__)
+                self._fetched = True
+                return
+        raise ClientException(
+            "r/{} does not have the rule {}".format(
+                self.subreddit, self.short_name
```

Comment:
Quite minor, but can we use `self.subreddit.display_name_prefixed`? In case it is a user subreddit, this will correctly prefix with "u/".
```suggestion
            "{} does not have the rule {}".format(
                self.subreddit.display_name_prefixed, self.short_name
```

Reply:
Shouldn't this be then propagated to all other classes that use subreddit display names?

Reply:
Off the top of my head, I don't know of other places where PRAW messages try to use a prefixed display name. I think that they mostly just print the subreddit name, sans prefix. So "AskReddit" in the case of a normal subreddit, or "u_spez" in the case of a user subreddit.

If there are cases where PRAW does this, it would be worth it just to update those.